### PR TITLE
Add hotplug/unplug memory in loop test

### PR DESCRIPTION
**Pull request by xutian**
This case can't be test with latest rhel guest, it block by kernel bug 1314306

### CODE REVIEW

File: qemu/tests/cfg/hotplug_mem_repeat.cfg [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Notes:
+#    Before start testing, please ensure systemd is your guest OS
+#    is support memory hotplug;
+- hotplug_memory_repeat:
+    type = hotplug_mem_repeat
```
Comment (CongSmile) [inline qemu/tests/cfg/hotplug_mem_repeat.cfg]: hotplug_mem ?

Comment (xutian) [inline qemu/tests/cfg/hotplug_mem_repeat.cfg]: oh, sorry. forget to add hotplug_mem_repeat.py. thanks your comment

**Comment by suqinhuang** [comment]
Hi xutian, 
what's the id

File: qemu/tests/cfg/hotplug_mem_repeat.cfg [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    only Linux
+    no Host_RHEL.6
```
Comment (spcui) [inline qemu/tests/cfg/hotplug_mem_repeat.cfg]: should be "no Host_RHEL.m6"

File: qemu/tests/hotplug_mem_repeat.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import time
```
Comment (CongLi) [inline qemu/tests/hotplug_mem_repeat.py]: Miss module "logging"

File: qemu/tests/hotplug_mem_repeat.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+try:
+    from virttest import error_context as step_engine
+except ImportError:
+    from autotest.client.shared.error import step_engine
+
+
+class MemoryHotplugRepeat(MemoryHotplugTest):
+
+    def start_test(self):
```
Comment (CongLi) [inline qemu/tests/hotplug_mem_repeat.py]: Please add doc string for this func().

File: qemu/tests/hotplug_mem_repeat.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            extra_params = (scalability_test and
+                            [{'slot_dimm': repeat}] or [None])[0]
+            step_engine.context(
+                "Hotplug/unplug loop '%d'" %
+                repeat, logging.info)
+            self.turn(vm, target_mem, extra_params)
+        vm.verify_alive()
+        vm.reboot()
+
+    def turn(self, vm, target_mem, extra_params=None):
```
Comment (CongLi) [inline qemu/tests/hotplug_mem_repeat.py]: Please add doc string for this func().

File: qemu/tests/hotplug_mem_repeat.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        memorys = self.get_all_memorys(vm)
```
Comment (CongLi) [inline qemu/tests/hotplug_mem_repeat.py]: 15:34:01 ERROR| AttributeError: 'MemoryHotplugRepeat' object has no attribute 'get_all_memorys'

**Comment by xutian** [comment]
deps on PR: https://github.com/avocado-framework/avocado-vt/pull/443/files

File: qemu/tests/hotplug_mem_repeat.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        :param vm: qemu target VM object
+        :param target_mem: memory name of target VM object
+        :param extra_params: params dict, that you want to update
+
+        """
+        memorys = self.get_all_memorys(vm)
+        if extra_params:
+            self.params.update(extra_params)
+        self.hotplug_memory(vm, target_mem)
+        memorys_added = self.get_all_memorys(vm) - memorys
```
Comment (CongLi) [inline qemu/tests/hotplug_mem_repeat.py]: 1:54:23 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem_repeat.py", line 47, in turn
11:54:23 ERROR|     memorys_added = self.get_all_memorys(vm) - memorys
11:54:23 ERROR| TypeError: unsupported operand type(s) for -: 'list' and 'list'

Comment (vivianQizhu) [inline qemu/tests/hotplug_mem_repeat.py]: met same issue.
13:32:55 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-37.0-py2.7.egg/avocado_vt/test.py", line 416, in _run_avocado
    raise test_exception
    
13:32:55 ERROR| TypeError: unsupported operand type(s) for -: 'list' and 'list'
    
13:32:55 ERROR| ERROR 01-smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u1.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.my-provider.hotplug_memory_repeat.backend_ram.policy_default -> TypeError: unsupported operand type(s) for -: 'list' and 'list'

Comment (xutian) [inline qemu/tests/hotplug_mem_repeat.py]: really a small bug in PR   https://github.com/avocado-framework/avocado-vt/pull/443. Hi @CongLi and @vivianQizhu would you like pull latest code and retest. 

Thanks,
Xu

File: qemu/tests/hotplug_mem_repeat.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        offline_memorys = self.get_offline_memorys(vm)
+        if not offline_memorys.issubset(memorys_added):
+            unexpected_memorys = offline_memorys - memorys_added
+            exceptions.TestFail(
+                "Unexpected offline memory %s" %
+                unexpected_memorys)
+        for memory in memorys_added:
+            self.memory_operate(vm, memory, 'online_movable')
```
Comment (CongLi) [inline qemu/tests/hotplug_mem_repeat.py]: 14:31:57 ERROR| ShellCmdError: Shell command failed: 'echo online_movable > /sys/devices/system/memory/memory72/state'    (status: 1,    output: 'echo online_movable > /sys/devices/system/memory/me<cho online_movable > /sys/devices/system/memory/mem                         \x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08ory72/state\n-bash: echo: write error: Invalid argument\n')

**Comment by xutian** [comment]
thanks @pezhang test results.

**Pull request merged**